scripts/bulk-update-status.py

```
@@ -0,0 +1,116 @@
+
+# Set the title to filter records by
+TITLE = None
+
+# Set the URLs for the API
+UPDATE_URL = "https://prexit.local/pluto-core/api/pluto/commission"
+GET_URL = "https://prexit.local/pluto-core/api/pluto/commission/list"
+
+STATUS_STRINGS = ["New", "Held", "Completed", "Killed", "In Production"]
+ALLOWED_INPUT = ["1", "2", "3", "4", "5", "6"]
+
+MAX_RECORDS_PER_PAGE = 100
+
+# get token from environment variable
+token = os.environ.get("PLUTO_TOKEN")
+
+headers = {
+    "Content-Type": "application/json",
+    "Authorization": f"Bearer {token}",
+}
+
+# Set up logging to write to a file
+logging.basicConfig(filename="data.log", level=logging.DEBUG)
+
+def get_filtered_records(timestamp, status, title=None) -> list:
+    request_body = { 
+    "match": "W_EXACT",
+    "completionDateBefore": timestamp,
+    "status": status
+    }
+    if title:
+        request_body['title'] = title
+
+    json_body = json.dumps(request_body)
+    records = []
+    try:
+        response = requests.put(GET_URL, headers=headers, data=json_body, verify=False)
+        response.raise_for_status()  # Raise an HTTPError if status is not 2xx
+        json_content = response.json()
+        total_records = json_content['count']
+        total_pages = (total_records + MAX_RECORDS_PER_PAGE - 1) // MAX_RECORDS_PER_PAGE
+        start_at = 0
+        for page in range(1, total_pages + 1):
+            print(f"loading page: {page}")
+            response = requests.put(f"{GET_URL}?startAt={start_at}&length={MAX_RECORDS_PER_PAGE}", data=json_body, headers=headers, verify=False)
+            response.raise_for_status()  # Raise an HTTPError if status is not 2xx
+            json_content = response.json()
+            logging.debug(f"page: {page}, records: {json_content['result']}")
+            records.extend(json_content['result'])
+            start_at += MAX_RECORDS_PER_PAGE
+
+    except requests.exceptions.RequestException as e:
+        print(e)
+    return records
```

Hmmm personally I would probably "let it crash" here rather than continuing with the records so far..... if the API is misbehaving I'd rather know about it!

The only exception to that (if you pardon the expression!) would be to have a retry-this-page operation on 503/502 response. Knowing pluto-core, though, this is so unlikely it's not worth worrying about

---

This is especially valid if your token happens to expire partway through the script run, come to think of it